### ISSUE TITLE
provide TypeScript definitions (.d.ts)

### ISSUE THREAD

**Issue by nippur72** [clr-format/clr-format#2]
I see the project is written in TypeScript, but there are no definitions (.d.ts files) in the npm package.

Would you please consider ship TypeScript definitions too?

**Comment by vmmitev** [comment]
Will do. But before that, since it's been a while that I worked on this project, so feel free to correct me.

In NPM the only exposed parts are:
```javascript
module.exports = String.format;
module.exports.setCulture = Format.setCulture;
module.exports.setCurrency = Format.setCurrency;
module.exports.Config = Format.Config;
```

You'd like a d.ts definition for this exposed API, right?

**Comment by nippur72** [comment]
> You'd like a d.ts definition for this exposed API, right?

yes, exactly. The most important reason is that you can import from typescript with:
```
import { format } from "clr-format";
```
without the error "module not found"

BTW, you might want to check the new TypeScript features, namely:
- `typings` can be specified directly in `package.json`
- optionally there is `export as namespace` to make a global-compatible typing
